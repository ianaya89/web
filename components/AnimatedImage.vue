<template>
  <img :src="src" :alt="alt" :class="{'clip-animation' : animationImg}" @load="animateImg">
</template>

<script>
export default {
  name: 'AnimatedImage',

  props: {
    src: {
      type: String,
      required: true
    },
    alt: {
      type: String,
      required: true
    }
  },

  data: () => ({
    animationImg: false
  }),

  methods: {
    animateImg() {
      this.animationImg = true
      setTimeout(() => {
        this.animationImg = false
      }, 1000)
    }
  }
}
</script>

<style lang="postcss" scoped>
img {
  @apply .absolute;
  object-fit: cover;
  z-index: -10;
}

@media (min-width: 577px) and (max-width: 991px) {
  img {
    @apply .h-full;
    right: -18rem;
  }
}

@media (max-width: 576px) {
  img {
    @apply .h-full;
    right: -16rem;
  }
}

.clip-animation {
  -webkit-clip-path: polygon(
    0 100%,
    0 100%,
    100% 100%,
    100% 100%,
    68% 100%,
    68% 100%,
    25% 100%,
    25% 100%
  );
  clip-path: polygon(
    0 100%,
    0 100%,
    100% 100%,
    100% 100%,
    68% 100%,
    68% 100%,
    25% 100%,
    25% 100%
  );
}

.clip-animation {
  animation: clip 1s 1;
}

@keyframes clip {
  0% {
    -webkit-clip-path: polygon(
      0 100%,
      0 100%,
      100% 100%,
      100% 100%,
      68% 100%,
      68% 100%,
      25% 100%,
      25% 100%
    );
    clip-path: polygon(
      0 100%,
      0 100%,
      100% 100%,
      100% 100%,
      68% 100%,
      68% 100%,
      25% 100%,
      25% 100%
    );
  }
  20% {
    -webkit-clip-path: polygon(
      0 33%,
      0 100%,
      100% 100%,
      100% 70%,
      68% 70%,
      68% 58%,
      25% 58%,
      25% 33%
    );
    clip-path: polygon(
      0 33%,
      0 100%,
      100% 100%,
      100% 70%,
      68% 70%,
      68% 58%,
      25% 58%,
      25% 33%
    );
  }
  100% {
    -webkit-clip-path: polygon(
      0 0,
      0 100%,
      100% 100%,
      100% 70%,
      68% 70%,
      68% 58%,
      25% 58%,
      25% 33%
    );
    clip-path: polygon(
      0 0,
      0 100%,
      100% 100%,
      100% 0,
      68% 0,
      68% 0,
      25% 0,
      25% 0
    );
  }
}
</style>
