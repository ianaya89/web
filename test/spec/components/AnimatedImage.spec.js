import { shallowMount } from '@vue/test-utils'
import AnimatedImage from '@/components/AnimatedImage'

jest.useFakeTimers()

describe('AnimatedImage', () => {
  it('renders default animated image', () => {
    const wrapper = shallowMount(AnimatedImage, {
      propsData: {
        src: 'some_image_url',
        alt: 'Image ale'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.vm.animateImg()

    // before the timeout runs out
    expect(wrapper.element).toMatchSnapshot()

    jest.runAllTimers()

    // after the timeout
    expect(wrapper.element).toMatchSnapshot()
  })
})
