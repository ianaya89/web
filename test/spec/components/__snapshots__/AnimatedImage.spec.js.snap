// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedImage renders default animated image 1`] = `
<img
  alt="Image ale"
  class=""
  src="some_image_url"
/>
`;

exports[`AnimatedImage renders default animated image 2`] = `
<img
  alt="Image ale"
  class=""
  src="some_image_url"
/>
`;

exports[`AnimatedImage renders default animated image 3`] = `
<img
  alt="Image ale"
  class=""
  src="some_image_url"
/>
`;
